### PR TITLE
Add Social Previews

### PR DESCRIPTION
**Pull request by kingthorin**
- Add hugo internal templates for twitter and Open Graph (Facebook etc) to `baseOf.html`
- Updated zapping-the-top-10 to have front matter details for social preview.

Unfortunately until this is live there's no good way to test it on social media sites.

Fixes zaproxy/zaproxy-website#45

https://gohugo.io/templates/internal/ was the primary resource for getting this setup, and should act as a reference going forward.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: site/content/blog/2012-10-22-zap-weekly-releases/index.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "ZAP Weekly Releases"
+description: "I've discussed this with the other ZAP developers, and we've decided to do weekly ZAP releases from the develop branch."
```
Comment (psiinon) [inline site/content/blog/2012-10-22-zap-weekly-releases/index.md]: Some of these summaries look a bit strange in isolation. Do we need them for old blog entries? Maybe just test with ZAPping the Top 10 and then add to new / current content?

File: site/content/docs/guides/zapping-the-top-10.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+- https://raw.githubusercontent.com/OWASP/Top10/master/2017/images/OWASP-Top-10.png
+tags: 
+- top10
+- guide
 type: page
 date: "2020-01-30"
 ---
 
-This document gives an overview of the automatic and manual components provided by the OWASP Zed Attack Proxy Project (ZAP) that are recommended for testing each of the OWASP Top Ten Project 2017 risks.
+This document gives an overview of the automatic and manual components provided by the OWASP Zed Attack Proxy (ZAP) that are recommended for testing each of the OWASP Top Ten Project 2017 risks.
```
Comment (psiinon) [inline site/content/docs/guides/zapping-the-top-10.md]: Either "the OWASP Zed Attack Proxy Project" (as before) or "OWASP Zed Attack Proxy Project" (no "the" or "project").

File: site/layouts/_default/baseof.html [hunk 3]
```diff
@@ -4,6 +4,10 @@
 <head>
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
+
+  {{ template "_internal/twitter_cards.html" . }}
+  {{ template "_internal/opengraph.html" . }}
```
Comment (psiinon) [inline site/layouts/_default/baseof.html]: Do the "_internal/twitter_cards.html" and "_internal/opengraph.html" need to be added? They dont seem to be part of the repo or this PR...

Comment (kingthorin) [reply site/layouts/_default/baseof.html]: Ya, their Hugo internal components.

File: site/content/blog/2012-10-22-zap-weekly-releases/index.md (hunk 1 above)
Comment (kingthorin) [reply site/content/blog/2012-10-22-zap-weekly-releases/index.md]: Sure, that's fine.

**Comment by kingthorin** [comment]
Feedback addressed.

File: site/content/docs/guides/zapping-the-top-10.md (hunk 2 above)
Comment (psiinon) [reply site/content/docs/guides/zapping-the-top-10.md]: Still think the "the" doesnt sound right :/ "provided by OWASP Zed Attack Proxy" sounds better to me, but not a biggie...

Comment (kingthorin) [reply site/content/docs/guides/zapping-the-top-10.md]: Missed it sorry, will tweak shortly

**Comment by kingthorin** [comment]
There we go. Sorry about the mix-up, for some reason I had it in my head that the change was in one of the reverted files so I didn't look back at it :(

**Comment by thc202** [comment]
Thanks!

**Pull request merged**